src/mds/Server.cc

```diff
@@ -6306,8 +6307,10 @@ void Server::handle_client_unlink(MDRequestRef& mdr)
   // lock
   set<SimpleLock*> rdlocks, wrlocks, xlocks;
 
-  for (int i=0; i<(int)trace.size()-1; i++)
-    rdlocks.insert(&trace[i]->lock);
+  for (auto &t : trace) {
+    rdlocks.insert(&t->lock);
```

Comment:
The new loop also adds `trace[trace.size()-1]` to `rdlocks`. Not quite the same loop :)

Reply:
nice catch -- guess I got a bit carried away :P